### DISCUSSION TITLE
Stupid node version warning in 1.1.13

### DISCUSSION THREAD

**Discussion by jacotec** [Ylianst/MeshCentral#5435] [General]
I've updated my server from version 1.1.10 to 1.1.13, here a warning pops up on the server screen:

`WARNUNG: MeshCentral will require Node v16 or above in the future, your current version is v14.21.3.`

But I'm already running node 18 on my server:

```
root@support:~# node -v
v18.14.2
```

Did anyone see the same issue?

**Comment by si458** [comment]
we added this feature because a lot of people where using old versions of nodejs and newer npm packages require at least the latest LTS version of nodejs

how did you install your meshcentral?
how do you start your meshcentral?

it could be a bug!?

**Reply by sysopdm** [reply]
Si458, if I update to version 1.1.14 and I have Node v10.19.0 installed, will the MeshCentral upgrade also upgrade Node or do I need to update Node first and then upgrade MC to 1.1.14?

**Reply by si458** [reply]
you need to stop meshcentral running first, then upgrade node to at least 12 (personally the latest list 18/20) then upgrade meshcentral!
Meshcentral will only upgrade itself, it won't upgrade node
(Unless u r using the docker image, then just already on node 18!)

**Reply by sysopdm** [reply]
Will give it a try and thank you!

**Reply by sysopdm** [reply]
Si458,
One last question, do you recommend going Node version 18 or 20 for MeshCentral?

**Reply by si458** [reply]
my personal option has always been use the latest LTS version 
which was 18 but then it was upgraded to 20 last week
18 is still in maintaince mode which should be fine! https://nodejs.org/en/download/releases

**Reply by sysopdm** [reply]
Thanks again!

**Comment by ubiquity60441** [comment]
I have run into the same issue.

I think it has to do with how nodejs was updated.

I thought I successfully updated to v18.18.0:

```
meshadmin@meshcentral:~$ node -v
v18.18.0
```

and as root:

```
root@meshcentral:~# node -v
v18.18.0
```

but when running `node -v` with sudo:

```
root@meshcentral:~# sudo node -v
v12.22.9
```

this is where my warning appears to be coming from.

**Reply by si458** [reply]
have you tried doing `which node` and `sudo which node` and check where its pointing node too?

**Reply by ubiquity60441** [reply]
both are pointing to /usr/local/bin/node

I just upgraded using `sudo -E env "PATH=$PATH" n stable` and now:

```
meshadmin@meshcentral:~$ sudo node -v
[sudo] password for meshadmin:
v18.18.2
```

but still getting warning, even after reboot.

**Reply by ubiquity60441** [reply]
does meshcentral check and use /usr/bin/node instead of /usr/local/bin/node?
If so, that seems to be where the problem is.

trying a brand new install on ubuntu server 22.04.2 following the instructions at https://ylianst.github.io/MeshCentral/install/install2/ and find that `node -v` shows v12.  So I upgrade using nvm but that does not update /usr/bin/node:

```
meshadmin@meshcentral2:~$ node -v
v18.18.2
meshadmin@meshcentral2:~$ /usr/bin/node -v
v12.22.9
```

**Reply by si458** [reply]
It depends how u run meshcentral on start up?
https://ylianst.github.io/MeshCentral/install/install2/#automatically-starting-the-server
If u use this method of the systemd, you just change where node is pointing too inside the service file and then reload systemd and restart server

**Reply by ubiquity60441** [reply]
okay, I think I got it working.

have to remove nodejs completely and add new repository.
steps I took for ubuntu server 22.04:

```
sudo systemctl stop meshcentral
sudo apt remove nodejs -y
sudo apt autoremove -y
sudo apt-get update
sudo apt-get install -y ca-certificates curl gnupg
sudo mkdir -p /etc/apt/keyrings
curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | sudo gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_18.x nodistro main" | sudo tee /etc/apt/sources.list.d/nodesource.list
sudo apt update
sudo apt install nodejs -y
sudo systemctl start meshcentral
```

**Reply by ubiquity60441** [reply]
> It depends how u run meshcentral on start up? https://ylianst.github.io/MeshCentral/install/install2/#automatically-starting-the-server If u use this method of the systemd, you just change where node is pointing too inside the service file and then reload systemd and restart server

okay, so you don't have to go through all I went through...just update /etc/systemd/system/meshcentral.service line that contains ExecStart=/usr/bin/node and change to ExecStart=/usr/**local**/bin/node
